projects/ADuCM3029_demo_cn0428_cn0429/src/cli.c

```diff
@@ -0,0 +1,986 @@
+	UART_TX("                                  <dur> = duration in ms, 10 to 200"_EOS);
+	UART_TX("readsensors                    -Read the ppb value of all sensors"_EOS);
+	UART_TX("setupdaterate <time>           -Configure update rate for stream mode"_EOS);
+	UART_TX("stopread                       -Stop sensor data reading"_EOS);
+	UART_TX("\r\n"_EOS);
+}
+
+/**
+   @brief Finds the next command line argument
+
+   @param args - pointer to the arguments on the command line.
+
+   @return pointer to the next argument.
+
+**/
+uint8_t *FindArgv(uint8_t *args)
+{
+	uint8_t *p = args;
+	int     fl = 0;
+
+	while (*p != 0) {
+		if ((*p == _SPC))
+			fl = 1;
+		else if (fl)
+			break;
+		p++;
+	}
+
+	return p;
+}
+
+/**
+   @brief Separates a command line argument
+
+    @param dst - pointer to a buffer where the argument will be copied
+    @param args - pointer to the current position of the command line .
+
+   @return none
+
+**/
+void GetArgv(char *dst, uint8_t *args)
+{
+	uint8_t  *s = args;
+	char     *d = dst;
+
+	while (*s) {
+		if (*s == _SPC)
+			break;
+		*d++ = *s++;
+	}
+
+	*d = '\0';
+}
+
+/**
+   @brief Puts command for FSM to the ring buffer
+
+   @param cmd - pointer to the command
+
+   @return none
+**/
+void CmdToRB(const char *cmd)
+{
+	uint8_t cmdLen = strlen(cmd);
+	for (int i = 0; i < cmdLen; i++) {
+		*RingBuf_Putvalue(&RX) = *cmd++;
+		RingBuf_Write_Increment(&RX);
+	}
+	Empty_Ring();
+}
+
+/**
+   @brief Command line process function
+
+   @return none
+**/
+void CmdProcess(void)
+{
+	cmdFunc   func;
+
+	/* <ENTER> key was pressed */
+	/* Find needed function based on typed command */
+	func = FindCommand((char *)cmdInString);
+
+	/* Check if there is a valid command */
+	if (func) {
+		/* Call the desired function */
+		(*func)(&cmdInString[2]);
+
+		/* Check if there is no match for typed command */
+	} else if (strlen((const char *)cmdInString) != 0) {
+		/* Display a message for unknown command */
+		UART_TX("Unknown command!"_EOS);
+	}
+	/* Prepare for next <ENTER> */
+	// uart_cmd = UART_FALSE;
+	CmdPrompt();
+}
+
+/**
+   @brief Command line prompt. Caller must verify that only the enabled board can run this.
+
+   @return int - UART status: UART_SUCCESS or error status
+**/
+void CmdPrompt(void)
+{
+//	int res;
+	static uint8_t count = 0;
+
+
+	/* Check first <ENTER> is pressed after reset */
+	if(count == 0) {
+		UART_TX("Welcome to CN0429!"_EOS);
+		UART_TX("Type <help> to see available commands..."_EOS);
+		count++;
+	}
+
+}
+
+/**
+   @brief Find available commands
+
+   @param cmd - command to search
+
+   @return cmdFunc - return the specific function for available command or NULL
+					 for invalid command
+**/
+cmdFunc FindCommand(char *cmd)
+{
+	cmdFunc func = NULL;
+	char *p = cmd;
+	int i = 0;
+	int f1 = 0;
+	int cmdlength = 0;
+
+	while(*p != 0) {
+		if (*p == _SPC)
+			f1 = 1;
+		else if (!f1)
+			cmdlength++;
+		p++;
+	}
+
+	while (CmdFun[i] != NULL) {
```

Comment:
regarding my comment here: https://github.com/analogdevicesinc/EVAL-ADICUP3029/pull/12/commits/4f5694854b122faead93c95e10cae42d6207dad4#r243188142  this infers that there is a correlation between the list of command strings and command functions ; 

this relation is not very strong, because they are 2 different arrays held together by a single function (this one), so if you were to change one array you'd break the other ;

Reply:
This is true. It's not a very strong relationship. But we are following the conventions from the example code in this repository on the ADuCM3029 CLI project and the cn0326 project.